.github/workflows/integration-tests-reusable.yml

```diff
@@ -110,15 +110,11 @@ jobs:
         unzip dafny/Package/CI.zip -d unzippedRelease
     - name: Load Z3
       if: "!inputs.all_platforms"
-      shell: pwsh
       run: |
-        Invoke-WebRequest https://github.com/Z3Prover/z3/releases/download/Z3-4.8.5/z3-4.8.5-x64-ubuntu-16.04.zip -OutFile z3.zip
-        Expand-Archive z3.zip .
-        Remove-Item z3.zip
-        Copy-Item z3-4.8.5-x64-ubuntu-16.04 dafny/Binaries/z3 -Recurse
-    - name: Set Z3 Permissions
-      if: "!inputs.all_platforms"
-      run: chmod +x dafny/Binaries/z3/bin/z3
+        sudo apt-get install -qq libarchive-tools
+        mkdir dafny/Binaries/z3
+        wget -qO- https://github.com/Z3Prover/z3/releases/download/Z3-4.8.5/z3-4.8.5-x64-ubuntu-16.04.zip | bsdtar -xf - -C dafny/Binaries/z3 --strip-components=1
```

Comment:
bsdtar supports unzip-ing?

Reply:
Yes, I was prototyping this on my Mac and was positively surprised as well, but then I was disappointed when I needed to install `libarchive-tools`.

Reply:
It's awkward that we `mkdir` first.